### PR TITLE
Convert guide to asciidoc markup format

### PR DESCRIPTION
**Pull request by oddhack**
@sfricke-samsung this is a WIP / proof of concept conversion of the entirety of the Guide to asciidoctor format. The motivation is the WG discussions about better unified documentation - in order to produce a unified document set with spec, guide, refpages, proposals, they all need to be in asciidoctor format.

The conversion was straightforward and has the feature that you can continue to either view the individual pages through the github previewer as with the current Markdown pages, or create a single document (currently checked in as guide.html, though we wouldn't publish it that way). It will need to be brought up to date with a few recent commits in master.

If we do this we'll have to handle redirections from existing deep links to .md files. We don't have access to .htaccess or other server-level rewrite mechanisms on github (AFAIK anyway), so probably the only viable way is to create a bunch of stub .md pages with links to the new .adoc pages. Eventually the .md pages could be retired.

### CODE REVIEW

**Comment by oddhack** [comment]
Added redirect .md pages - see https://github.com/KhronosGroup/Vulkan-Guide/blob/adoc-convert/chapters/atomics.md for an example.

**Comment by oddhack** [comment]
@sfricke-samsung this is ready to go from my perspective, let me know what you need improved before migrating over to asciidoctor.

**Review by sfricke-samsung — APPROVED** [review]
@oddhack Looks good to me too, made a slight wording change to redirect pages

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,265 @@
+// Copyright 2019-2021 The Khronos Group, Inc.
+// SPDX-License-Identifier: CC-BY-4.0
+
+= Vulkan^®^ Guide
+:regtitle: pass:q,r[^®^]
+The Khronos{regtitle} Vulkan Working Group
+:data-uri:
```
Comment (sfricke-samsung) [inline README.adoc:7]: is there a reason for the ToC in the `README.adoc` since there is already one in the `Guide.adoc`?

Comment (oddhack) [reply README.adoc:7]: Removed it. In the github viewer it's useless as the ToC is inline with the rest of the document which is itself basically a ToC. In a different renderer like Asciidoctor.js online preview extension, you get a nice sidebar ToC, but that's not the usual use case.

**Comment by oddhack** [comment]
@sfricke-samsung I believe this can be merged at your convenience. I removed the ToC from README.adoc, migrated from Travis to Github Actions for CI, and fixed CI.

BTW, what the heck is the provenance of 'awesome_bot'? I sorta see what it *does*, but trying to find out what 'awesome projects' are per the github repository quickly led me down a rathole with no apparent central 'awesome project' definition and really just seemed to be a random collection of Markdown READMEs. It does seem to (mostly) work on asciidoctor markup as well, though had to add one whitelist exception for a URL with '()' in it.

**Comment by sfricke-samsung** [comment]
@oddhack So never touched Github Actions (or much of travis CI really) so thanks for converting it over!
The "awesome" project is just a thing on GitHub where people go listing a bunch of links for a given topic, I came across the awesome_bot repo in the past as a simple tool to add to GitHub CI to check when URL are dead... it makes sure this Guide doesn't have broken URLs, it [caught a bunch of spots](https://github.com/KhronosGroup/Vulkan-Guide/pull/131/commits/40b194a1032ccf1f66269e8b7b535a112e684d8b) where we changed `.md` to `.adoc` for other repo URL. Also saw the stupid wiki URL with the `( )` in it, good idea just whitelisting it

As for clicking "merge" just going to get final "no objections" sign off from WG this week

**Comment by KhronosKris** [comment]
There is currently a project in progress to convert the Guide into Japanese. These translations are coming out in Markdown. 

Will we be able to easily convert them to Asciidoc after the fact or should we wait before continuing on more translations?

**Comment by oddhack** [comment]
> There is currently a project in progress to convert the Guide into Japanese. These translations are coming out in Markdown.
> 
> Will we be able to easily convert them to Asciidoc after the fact or should we wait before continuing on more translations?

The asciidoc conversion is not difficult. I did it in a couple of hours with some editor macros, much of the effort a one-time thing to learn how to write links in a way that both the separate .adoc files render correctly on github, and the combined guide.adoc -> guide.html process correctly when building offline. It is possible to sensibly compare the .md (before conversion) and .adoc files to see how the markup changed. If the conversion can be signed off in the near future, I think would be better for the translators to work it into their flow if they're able to do so. I could *probably* convert them myself after they're done, though I have zero Japanese language knowledge beyond a few words from my time there when my Dad was in the Air Force. 

That said my experience using asciidoc markup for anything other than English is nonexistent and writing a short test document using all the markup features in the Guide (offhand: bullet lists, titles, source highlighting, NOTE blocks, italic / bold emphasis, images, internal and external links, probably a few minor things) would probably be advisable.

**Comment by KhronosKris** [comment]
My gut is telling me that it might be best to pause the translations and continue again once the guide has moved to AD, just to avoid extra unnecessary conversion work. This also gives us time to investigate how well Japanese and AD work together.

**Comment by TakashiKhronos** [comment]
As long as I quickly checked about Japanese support in asciidoc, it seems format itself support Japanese, but there is a known issue for PDF conversion (Japanese font is missing now)

**Comment by oddhack** [comment]
> As long as I quickly checked about Japanese support in asciidoc, it seems format itself support Japanese, but there is a known issue for PDF conversion (Japanese font is missing now)

Thank you! The Guide is not currently published in PDF format, so if HTML conversion works that may be sufficient.

**Pull request merged**